src/main/java/net/neoforged/neoforge/fluids/FluidType.java

```diff
@@ -372,6 +374,17 @@ public boolean canHydrate(Entity entity) {
         return this.canHydrate;
     }
 
+    /**
+     * Returns whether a fluid should be treated like an entity is residing in water.
+     *
+     * <p> This sets {@link Entity#isInWater()} to return true, which is used for things like aquatic animal swimming, slowed minecart movement, and conduit functionality.
+     *
+     * @return {@code true} if this fluid acts like water, {@code false} otherwise
+     */
+    public boolean actsLikeWater() {
```

Comment:
Doesn't this break lava? Like doesn't vanilla lava use updateFluidHeightAndDoFluidPushing as well for pushing and sloping the top based on nearby fluids/blocks

This might be better called something else if lava is using this

Reply:
`updateFluidHeightAndDoFluidPushing` does change the height and whatnot, yes, but the boolean the method returns is only used for updating the `wasTouchingWater` flag in the vanilla codebase. If other mods potentially used it, the vanilla method would return true if the fluid was in the specified tag, which is initially why I wanted to go the tag route first. 

Keeping it functioning like vanilla's did so mods could utilize it would probably require a tag-based solution, which I was told to stay away from. But if we're only accounting for vanilla here this works as intended